Loop/Utilities/TranslationLayer.swift

```diff
@@ -0,0 +1,93 @@
+//
+//  TranslationLayer.swift
+//  Loop
+//
+//  Created by Kami on 8/7/2024.
+//
+
+import AppKit
+import Defaults
+import Foundation
+
+/// Represents a keyboard shortcut configuration for a Rectangle action.
+struct RectangleShortcut: Codable {
+    let keyCode: Int
+    let modifierFlags: Int
+}
+
+/// Represents the configuration of Rectangle app shortcuts.
+struct RectangleConfig: Codable {
+    let shortcuts: [String: RectangleShortcut]
+}
+
+/// Translates the RectangleConfig to an array of WindowActions for Loop.
+/// - Parameter rectangleConfig: The RectangleConfig instance to translate. (this works w/ both normal and pro)
+/// - Returns: An array of WindowAction instances corresponding to the RectangleConfig.
+func translateRectangleConfigToWindowActions(rectangleConfig: RectangleConfig) -> [WindowAction] {
```

Comment:
Avoid top-level functions, probably makes sense to put all of these inside a `RectangleTranslationLayer` enum